src/Ganeti/Query/Exec.hs

```diff
@@ -88,6 +96,21 @@ rethrowAnnotateIOError :: String -> IO a -> IO a
 rethrowAnnotateIOError desc =
   modifyIOError (\e -> annotateIOError e desc Nothing Nothing)
 
+-- | Process 1.6.3. introduced the getPid function, for older versions
+-- provide an implemention here (https://github.com/haskell/process/pull/109)
+
+getPid' :: ProcessHandle -> IO (Maybe Pid)
+#if MIN_VERSION_process(1,6,3)
```

Comment:
I am a bit reluctant to add this to 3.0, but then again I don't know how many people will be building from source on 18.04. Let's say we go for it, then it should go into Ganeti.Compat instead, to keep the code in Ganeti.Query.Exec cleaner.

Reply:
> Let's say we go for it, then it should go into Ganeti.Compat instead, to keep the code in Ganeti.Query.Exec cleaner.

I've tried to move the function definition into Ganeti.Compat. I hope it's not that far off.

Reply:
> I am a bit reluctant to add this to 3.0, but then again I don't know how many people will be building from source on 18.04.

I don't know either. But this change seems small enough, to not support it. Do you see possible future troubles, that may arise from depending on ghc-8.0.2 (the bionic version)? Or could other old software break functionality, like qemu-2.11 (i.e. WRT # #1433)? Having a qa-suite for 18.04 would be fine.